pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -135,28 +136,28 @@ func (cb *ClusterBuilder) applyDestinationRule(proxy *model.Proxy, cluster *apiv
 }
 
 // buildDefaultCluster builds the default cluster and also applies default traffic policy.
-func (cb *ClusterBuilder) buildDefaultCluster(name string, discoveryType apiv2.Cluster_DiscoveryType,
+func (cb *ClusterBuilder) buildDefaultCluster(name string, discoveryType cluster.Cluster_DiscoveryType,
 	localityLbEndpoints []*endpoint.LocalityLbEndpoints, direction model.TrafficDirection,
-	port *model.Port, meshExternal bool) *apiv2.Cluster {
-	cluster := &apiv2.Cluster{
+	port *model.Port, meshExternal bool) *cluster.Cluster {
+	c := &cluster.Cluster{
 		Name:                 name,
-		ClusterDiscoveryType: &apiv2.Cluster_Type{Type: discoveryType},
+		ClusterDiscoveryType: &cluster.Cluster_Type{Type: discoveryType},
 	}
 
 	switch discoveryType {
-	case apiv2.Cluster_STRICT_DNS:
-		cluster.DnsLookupFamily = apiv2.Cluster_V4_ONLY
+	case cluster.Cluster_STRICT_DNS:
+		c.DnsLookupFamily = cluster.Cluster_V4_ONLY
 		dnsRate := gogo.DurationToProtoDuration(cb.push.Mesh.DnsRefreshRate)
-		cluster.DnsRefreshRate = dnsRate
-		cluster.RespectDnsTtl = true
+		c.DnsRefreshRate = dnsRate
+		c.RespectDnsTtl = true
 		fallthrough
-	case apiv2.Cluster_STATIC:
+	case cluster.Cluster_STATIC:
 		if len(localityLbEndpoints) == 0 {
-			cb.push.AddMetric(model.DNSNoEndpointClusters, cluster.Name, cb.proxy,
-				fmt.Sprintf("%s cluster without endpoints %s found while pushing CDS", discoveryType.String(), cluster.Name))
+			cb.push.AddMetric(model.DNSNoEndpointClusters, c.Name, cb.proxy,
+				fmt.Sprintf("%s cluster without endpoints %s found while pushing CDS", discoveryType.String(), c.Name))
 			return nil
 		}
-		cluster.LoadAssignment = &apiv2.ClusterLoadAssignment{
+		c.LoadAssignment = &endpoint.ClusterLoadAssignment{
```

Comment:
Not understanding how this works, I will just point out that this particular change looks a little suspect. Why did line 159 use apiv2.ClusterLoadAssignment rather than an endpoint.ClusterloadAsisgment? Was endpoint.ClusterLoadAssignment not set properly in v2 of XDS?

Reply:
Yes they changed the package structure